src/app/qgsmaptooloffsetcurve.cpp

```diff
@@ -346,6 +347,30 @@ void QgsMapToolOffsetCurve::applyOffset( double offset, Qt::KeyboardModifiers mo
 
   destLayer->beginEditCommand( tr( "Offset curve" ) );
 
+  QgsAvoidIntersectionsOperation avoidIntersections;
+
+  connect( &avoidIntersections, &QgsAvoidIntersectionsOperation::messageEmitted, this, &QgsMapTool::messageEmitted );
+
+  const QSet<QgsFeatureId> ignoredFeatures = ( QgsProject::instance()->avoidIntersectionsMode() != Qgis::AvoidIntersectionsMode::AllowIntersections && ( modifiers & Qt::ControlModifier ) ) ?
```

Comment:
Is `QgsProject::instance()->avoidIntersectionsMode() != Qgis::AvoidIntersectionsMode::AllowIntersections`  really needed because if mode allows intersection the apply() method would do nothing and just return

Reply:
Ah, no it's not. That was an oversight.